sdks/python/apache_beam/testing/load_tests/load_test_metrics_utils.py

```diff
@@ -36,49 +47,79 @@
   SchemaField = None
   NotFound = None
 
-RUNTIME_LABEL = 'runtime'
-SUBMIT_TIMESTAMP_LABEL = 'submit_timestamp'
-
-
-def _get_schema_field(schema_field):
+RUNTIME_METRIC = 'runtime'
+COUNTER_LABEL = "total_bytes_count"
+
+ID_LABEL = 'test_id'
+SUBMIT_TIMESTAMP_LABEL = 'timestamp'
+METRICS_TYPE_LABEL = 'metric'
+VALUE_LABEL = 'value'
+
+SCHEMA = [
+    {'name': ID_LABEL,
+     'type': 'STRING',
+     'mode': 'REQUIRED'
+    },
+    {'name': SUBMIT_TIMESTAMP_LABEL,
+     'type': 'TIMESTAMP',
+     'mode': 'REQUIRED'
+    },
+    {'name': METRICS_TYPE_LABEL,
+     'type': 'STRING',
+     'mode': 'REQUIRED'
+    },
+    {'name': VALUE_LABEL,
+     'type': 'FLOAT',
+     'mode': 'REQUIRED'
+    }
+]
+
+
+def get_schema_field(schema_field):
   return SchemaField(
       name=schema_field['name'],
       field_type=schema_field['type'],
       mode=schema_field['mode'])
 
 
+def get_element_by_schema(schema_name, insert_list):
+  for element in insert_list:
+    if element['label'] == schema_name:
+      return element['value']
+
+
 class BigQueryClient(object):
-  def __init__(self, project_name, table, dataset, schema_map):
+  def __init__(self, project_name, table, dataset):
     self._namespace = table
 
     self._bq_client = bigquery.Client(project=project_name)
 
-    schema = self._parse_schema(schema_map)
-    self._schema_names = self._get_schema_names(schema)
-    schema = self._prepare_schema(schema)
+    self._schema_names = self._get_schema_names()
+    schema = self._prepare_schema()
 
     self._get_or_create_table(schema, dataset)
 
-  def match_and_save(self, result_list):
-    rows_tuple = tuple(self._match_inserts_by_schema(result_list))
-    self._insert_data(rows_tuple)
+  def match_and_save(self, results_lists):
```

Comment:
Could you document what type `results_list` is?
It seems that each item in `results_list` is a list of dictionaries, and each dict looks like:
`{'label': SUBMIT_TIMESTAMP_LABEL, 'value': time.time()}`
but I'm not 100% sure.

I think this module would be easier to understand if each item in `results_list` was a single dict:
```py
{
  ID_LABEL: uuid,
  SUBMIT_TIMESTAMP_LABEL: time.time(),
  METRICS_TYPE_LABEL: RUNTIME_METRIC,
  VALUE_LABEL: value,
}
```

Note that `_bq_client.insert_rows()` also accepts a list of dicts so there would be no need to convert the above to tuple form.

Reply:
I agree with you in 100%. I had same impression this naming is not so clear. I will refactor it according to suggestions. Hopefully it will simplify.